libraries/CryptoInterface/src/CryptoInterface.h

```diff
@@ -0,0 +1,806 @@
+/*
+ * BearSSL Copyright (c) 2016 Thomas Pornin <pornin@bolet.org>
+ * Rest of this file Copyright (C) 2019 Anders Löfgren
+ *
+ * License (MIT license):
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef __ESP8266ARDUINOCRYPTOINTERFACE_H__
+#define __ESP8266ARDUINOCRYPTOINTERFACE_H__
+
+#include <Arduino.h>
+
+namespace experimental
+{
+  namespace CryptoInterface 
+  {
+    /**
+     * Regarding constant-time (CT) HMAC:
+     * 
+     * Basically, constant-time algorithms makes it harder for attackers to learn things about your system based on the execution time of code. 
+     * Good intro here: https://www.bearssl.org/constanttime.html
+     * 
+     * It should be noted that every HMAC is already partially constant-time. Quoting the link above: 
```

Comment:
+1 here and elsewhere for the detailed comments!

Reply:
Thanks!